src/openzeppelin/tests/test_erc721.cairo

```diff
@@ -0,0 +1,475 @@
+use openzeppelin::introspection::erc165;
+use openzeppelin::token::erc721;
+use openzeppelin::token::erc721::ERC721;
+
+use starknet::contract_address_const;
+use starknet::ContractAddress;
+use starknet::testing::set_caller_address;
+use integer::u256;
+use integer::u256_from_felt252;
+use zeroable::Zeroable;
+
+const NAME: felt252 = 111;
+const SYMBOL: felt252 = 222;
+const URI: felt252 = 333;
+
+fn TOKEN_ID() -> u256 {
+    u256_from_felt252(7)
+}
+
+fn ZERO() -> ContractAddress {
+    contract_address_const::<0>()
+}
+fn OWNER() -> ContractAddress {
+    contract_address_const::<1>()
+}
+fn RECIPIENT() -> ContractAddress {
+    contract_address_const::<2>()
+}
+fn SPENDER() -> ContractAddress {
+    contract_address_const::<3>()
+}
+fn OPERATOR() -> ContractAddress {
+    contract_address_const::<4>()
+}
+fn OTHER() -> ContractAddress {
+    contract_address_const::<5>()
+}
+
+///
+/// Setup
+///
+
+fn setup() {
+    ERC721::initializer(NAME, SYMBOL);
+    ERC721::_mint(OWNER(), TOKEN_ID());
+}
+
+///
+/// Initialize
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_initialize() {
+    ERC721::initializer(NAME, SYMBOL);
+
+    assert(ERC721::name() == NAME, 'Name should be NAME');
+    assert(ERC721::symbol() == SYMBOL, 'Symbol should be SYMBOL');
+    assert(ERC721::balance_of(OWNER()) == u256_from_felt252(0), 'Should be zero');
+
+    assert(ERC721::supports_interface(erc721::IERC721_ID), 'missing interface ID');
```

Comment:
```suggestion
    assert(ERC721::supports_interface(erc721::IERC721_ID), 'Missing interface ID');
```
we should settle on a style for error messages

Reply:
sorry this was wrong. ERC165 uses lowercase. we need indeed to settle on a style and apply it all over, probably it can wait for another issue.

Reply:
https://github.com/OpenZeppelin/cairo-contracts/issues/595